connectors/src/lib/temporal_monitoring.ts

```diff
@@ -0,0 +1,89 @@
+import { Context } from "@temporalio/activity";
+import {
+  ActivityExecuteInput,
+  ActivityInboundCallsInterceptor,
+  Next,
+} from "@temporalio/worker";
+import StatsDClient from "hot-shots";
+
+import logger, { Logger } from "@connectors/logger/logger";
+
+import { WorkflowErrorType } from "./error";
+
+/** An Activity Context with an attached logger */
+export interface ContextWithLogger extends Context {
+  logger: typeof logger;
+}
+
+export const statsDClient = new StatsDClient({});
+
+export class ActivityInboundLogInterceptor
+  implements ActivityInboundCallsInterceptor
+{
+  public readonly logger: Logger;
+  private readonly context: Context;
+
+  constructor(ctx: Context, logger: Logger) {
+    this.context = ctx;
+    this.logger = logger.child({
+      activityName: ctx.info.activityType,
+      workflowName: ctx.info.workflowType,
+      workflowId: ctx.info.workflowExecution.workflowId,
+      workflowRunId: ctx.info.workflowExecution.runId,
+      activityId: ctx.info.activityId,
+    });
+
+    // Set a logger instance on the current Activity Context to provide
+    // contextual logging information to each log entry generated by the Activity.
+    (ctx as ContextWithLogger).logger = this.logger;
+  }
+
+  async execute(
+    input: ActivityExecuteInput,
+    next: Next<ActivityInboundCallsInterceptor, "execute">
+  ): Promise<unknown> {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let error: any = undefined;
+    const startTime = new Date();
+    const tags = [
+      `activity_name:${this.context.info.activityType}`,
+      `workflow_name:${this.context.info.workflowType}`,
+      `activity_id:${this.context.info.activityId}`,
+      `workflow_id:${this.context.info.workflowExecution.workflowId}`,
+      `workflow_run_id:${this.context.info.workflowExecution.runId}`,
+    ];
+
+    try {
+      return await next(input);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (err: any) {
+      error = err;
+      throw err;
+    } finally {
+      const durationMs = new Date().getTime() - startTime.getTime();
+      if (error) {
+        const errorType: WorkflowErrorType = error.type
+          ? error.type
+          : "unhandled_internal_activity_error";
+
+        this.logger.error(
+          {
+            durationMs: durationMs,
+            error: {
+              type: errorType,
```

Comment:
I agree, but most errors have a .message field, so I was trying to display it.
I will do what you suggested, hopefully it will be serializable by the logger. I think I have seen some errors objects exposing a message field that was not serialized by the logger in the past, hence my initial code. We'll see live.